cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityStatusStore.java

```diff
@@ -0,0 +1,209 @@
+   */
+  @Override
+  public void addOrUpdateCapability(String capability, CapabilityStatus status,
+                                    CapabilityConfig config) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.CapabilitiesStore.CAPABILITIES);
+      Collection<Field<?>> fields = new ArrayList<>();
+      fields.add(Fields.stringField(StoreDefinition.CapabilitiesStore.NAME_FIELD, capability));
+      fields.add(Fields.stringField(StoreDefinition.CapabilitiesStore.STATUS_FIELD, status.name().toLowerCase()));
+      fields.add(Fields.stringField(StoreDefinition.CapabilitiesStore.CONFIG_FIELD, GSON.toJson(config)));
+      fields.add(Fields.longField(StoreDefinition.CapabilitiesStore.UPDATED_TIME_FIELD, System.currentTimeMillis()));
+      capabilityTable.upsert(fields);
+    }, IOException.class);
+  }
+
+  /**
+   * Delete capability
+   *
+   * @param capability
+   * @throws IOException
+   */
+  @Override
+  public void deleteCapability(String capability) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.CapabilitiesStore.CAPABILITIES);
+      Collection<Field<?>> fields = new ArrayList<>();
+      fields.add(Fields.stringField(StoreDefinition.CapabilitiesStore.NAME_FIELD, capability));
+      capabilityTable.delete(fields);
+    }, IOException.class);
+  }
+
+  /**
+   * Adds or update capability operations
+   *
+   * @param capability
+   * @param actionType
+   * @param config
+   * @throws IOException
+   */
+  public void addOrUpdateCapabilityOperation(String capability, CapabilityAction actionType,
+                                             CapabilityConfig config) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.CapabilitiesStore.CAPABILITY_OPERATIONS);
+      Collection<Field<?>> fields = new ArrayList<>();
+      fields.add(Fields.stringField(StoreDefinition.CapabilitiesStore.NAME_FIELD, capability));
+      fields.add(Fields.stringField(StoreDefinition.CapabilitiesStore.ACTION_FIELD, actionType.name().toLowerCase()));
+      fields.add(Fields.stringField(StoreDefinition.CapabilitiesStore.CONFIG_FIELD, GSON.toJson(config)));
+      capabilityTable.upsert(fields);
+    }, IOException.class);
+  }
+
+  /**
+   * Deletes capability operations
+   *
+   * @param capability
+   * @throws IOException
+   */
+  public void deleteCapabilityOperation(String capability) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.CapabilitiesStore.CAPABILITY_OPERATIONS);
+      Collection<Field<?>> fields = new ArrayList<>();
+      fields.add(Fields.stringField(StoreDefinition.CapabilitiesStore.NAME_FIELD, capability));
```

Comment:
Same here. You can use Collections.singleton

Reply:
Done